keras_cv/training/contrastive/contrastive_trainer.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+
+class ContrastiveTrainer(keras.Model):
+    """Creates a self-supervised contrastive trainer for a model.
+
+    Args:
+        encoder: a `keras.Model` to be pre-trained. In most cases, this encoder
+            should not include a top dense layer.
+        augmenter: a preprocessing layer to randomly augment input images for contrastive learning,
+            or a tuple of two separate augmenters for the two sides of the contrastive pipeline.
+        projector: a projection model for contrastive training, or a tuple of two separate
+            projectors for the two sides of the contrastive pipeline.
+        include_probe: Whether to include a single fully-connected layer during
```

Comment:
lets rewrite this according to Francois comment on the doc to keep this generalize-able.  `probe` - can be a model, `None`

Reply:
an added benefit of this is it makes it easier for users to know whether or not to set `from_logits=True` on probe loss.

Reply:
Agreed -- was just waiting on doc review to go ahead with this but I think it's the right approach so I've done this.